lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -4308,6 +4308,147 @@ class ConvertAtenIndexTensorOp : public OpConversionPattern<AtenIndexTensorOp> {
 };
 } // namespace
 
+// Helper function for creating FuncOp given function name and function type.
+static FuncOp findOrCreateGeneratorFunc(OpBuilder &b, Operation *op,
+                                        StringRef fnName, FunctionType fnTy) {
+  auto module = op->getParentOfType<ModuleOp>();
+  auto funcOp = module.lookupSymbol<FuncOp>(fnName);
+  if (funcOp)
+    return funcOp;
+
+  OpBuilder::InsertionGuard guard(b);
```

Comment:
TorchToLinalg is a FuncOp pass, so it cannot mutate the outer module like this. I would recommend adding an op `torch_c.get_next_seed : () -> i64` and then have InsertRngGlobals basically create a function `getNextSeed` (signature () -> i64) and a global `currentSeed` (one-element memref of i64).